### ISSUE TITLE
Implement New Metric Type for Constant Value Metrics to Optimize Storage in Prometheus

### ISSUE THREAD

**Issue by rajkumarkandasami** [prometheus/prometheus#13588]
### Introduction:
I want to start a discussion on integration of a new metric type within Prometheus. The proposed metric type can help optimize storage and processing for  gauge metric with unchanged/redundant value.

### Motivation:
Values of Metrics such as `prometheus_build_info, kube_persistentvolume_info, kube_pod_container_resource_limits, kube_pod_container_resource_requests` rarely undergo changes.
The repetitive storage of these unchanging values poses unnecessary storage and processing burdens. 

### Description:
The proposed metric type would retain a simplified dataset crucial for monitoring and alerting.
It contains the metric name, labels, value, start time, and end time(last scrape time with unchanged value).
Despite the minimalist approach to storage, In query time the metric will be assumed/behave of a gauge type. Durig query this metric will have X value between start and end time, with N data points for N seconds. 

### Community Feedback:
I deeply value community input in refining and validating this proposal. I sincerely welcome comments, suggestions, and critiques to ensure that the proposed metric type resonates with the community's requisites and expectations.

### Conclusion:
In conclusion, I firmly believe that introducing a new metric type that mirrors gauge functionality while optimizing storage and processing for metrics would significantly  benefit Prometheus users. I am grateful for the opportunity to present this proposal and eagerly anticipate collaborative discussions to refine and implement this feature.

**Comment by rajkumarkandasami** [comment]
If we don't want to introduce a new metric type, may be we can consider adding some flag to gauge metric and keep only first and last data points if flag is on. On query time we virtually add the missing data points.
